crates/ide/src/folding_ranges.rs

```diff
@@ -109,6 +111,15 @@ pub(crate) fn folding_ranges(file: &SourceFile) -> Vec<Fold> {
                         res.push(Fold { range, kind: FoldKind::Statics })
                     }
                 }
+
+                // Fold where clause
+                if node.kind() == WHERE_CLAUSE && !visited_where_clauses.contains(&node) {
+                    if let Some(range) =
+                        contiguous_range_for_where(&node, &mut visited_where_clauses)
+                    {
+                        res.push(Fold { range, kind: FoldKind::WhereClause })
+                    }
+                }
```

Comment:
These cases are for consequtive items of the same type, a where clause has an encompassing node, so this logic can be removed and instead a single new case added to `fold_kind`

Reply:
But when we fold the where clause as a whole, it hides the `where` keyword

```rust
fn foo()
where
    A: Foo,
    B: Foo,
    C: Foo,
    D: Foo, {}
```
will be:

```rust
fn foo()
/* */ {}
```

instead of 

```rust
fn foo()
where /* */ {}
```
and i won't be able to check if it's a multi-line where clause or not, am I missing something ?